src/test/java/io/pinecone/clients/ConnectionsMapTest.java

```diff
@@ -0,0 +1,118 @@
+package io.pinecone.clients;
+
+import io.pinecone.configs.PineconeConfig;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.openapitools.client.ApiException;
+import org.openapitools.client.api.ManageIndexesApi;
+import org.openapitools.client.model.IndexModel;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class ConnectionsMapTest {
+
+    @Test
+    public void testMultiplePineconeInstancesForIndexes() {
+        Pinecone pinecone1 = mock(Pinecone.class);
+        Pinecone pinecone2 = mock(Pinecone.class);
+
+        assertNotSame(pinecone1, pinecone2);
+
+        // Tests for blocking indexes
+        Index index1 = mock(Index.class);
+        Index index2 = mock(Index.class);
+
+        when(pinecone1.getIndexConnection("index1")).thenReturn(index1);
+        when(pinecone2.getIndexConnection("index2")).thenReturn(index2);
+
+        Index resultIndex1 = pinecone1.getIndexConnection("index1");
+        Index resultIndex2 = pinecone2.getIndexConnection("index2");
+
+        verify(pinecone1).getIndexConnection("index1");
+        verify(pinecone2).getIndexConnection("index2");
+        assertNotSame(resultIndex1, resultIndex2);
+    }
```

Comment:
I'm not really clear what is being tested in this method, since everything is being mocked, even the method that is being tested? 

Seems like we need to mock `getIndexHost()` and potentially `getConnection`, but then otherwise we should let the actual `getIndexConnection` method run.

Reply:
Updated the test and in future, we will like to tests for the contents as well